velox/type/SimpleFunctionApi.h

```diff
@@ -468,14 +468,14 @@ template <>
 struct MaterializeType<Varchar> {
   using nullable_t = std::string;
   using null_free_t = std::string;
-  static constexpr bool requiresMaterialization = false;
+  static constexpr bool requiresMaterialization = true;
```

Comment:
Dont we need a materialize() for the element type if this is set to true ? Is that handled somewhere ?
Can you write a brief test ?